### PR TITLE
Add building on Windows and USB booting guide

### PR DESCRIPTION
**Pull request by redstonee**
The code may be piece of shit
OE~12X%9NIYUD45(OB}W{1F
65LJU 0O21QJL@LNGBWR}9A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
+.vscode
 build/
 test/
 tools/*.bin
-mksunxi
\ No newline at end of file
+mksunxi
+mksunxi.exe
```
Comment (YuzukiTsuru) [inline .gitignore:6]: Windows use precompiled mksunxi.exe

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
+lib/arduino
```
Comment (YuzukiTsuru) [inline .gitignore:7]: separate arduino support in a new branch, syterkit support build as library in v0.2.1

File: CMakeLists.txt [hunk 3]
```diff
@@ -2,20 +2,16 @@
 
 cmake_minimum_required(VERSION 3.10)
 
+# Set the toolchain location if using Windows
+set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_LIST_DIR}/cmake/toolchain-arm-none-eabi.cmake)
```
Comment (YuzukiTsuru) [inline CMakeLists.txt:6]: Do not harddefine the compiler to use in syterkit

**Review by YuzukiTsuru — COMMENTED** [review]
Syterkit is not limited to the compiler used, which makes it easier to integrate it into various versions of the compilation framework, for the Windows platform compilation, my recommendation is
1. Use pre-compiled mksunxi
2. Use Windows proprietary macros to define and automatically specify the compiler

For Arduino, due to the characteristics of Arduino's directory framework, SyterKit does not include these operations. It is recommended that SyterKit be referenced as a library, which will be the modification direction of the next version of Syterkit

**Comment by redstonee** [comment]
好家伙真打算整Arduino了🥰🥰🥰🥰🥰

File: .gitignore (hunk 1 above)
Comment (redstonee) [reply .gitignore:6]: 🉑

**Comment by redstonee** [comment]
Committed more shit

**Comment by YuzukiTsuru** [comment]
Due to the merged version upgrade, the schema is modified and needs to be re-modified

**Pull request closed without merging**